### ISSUE TITLE
Serialization exceptions not handled.

### ISSUE THREAD

**Issue by admizh** [reportportal/reportportal#15]
We use corporate proxy. And our proxy not allow send any request with RP. I will check it in our network side. But any way report portal not handle this issue:
For understanding:
When suite start RP call reportPortalService.startRootTestItem(rq);
Our proxy return internal html page with status: 503.
![image](https://cloud.githubusercontent.com/assets/6014483/20713308/41cdc486-b661-11e6-9cd6-8b828ddb6fcb.png)

 And RP catched it:
![image](https://cloud.githubusercontent.com/assets/6014483/20712826/3798d89a-b65f-11e6-87d8-214e71f3d905.png)
But method handleException(...) is silent.  And i not see any problems in my console.
![image](https://cloud.githubusercontent.com/assets/6014483/20713464/d284a7ba-b661-11e6-9c62-095a0b2d6794.png)

Because com.google.common.base.Throwables#throwIfUnchecked - not handle "Serialize" exceptions type by default.

**Comment by avarabyeu** [comment]
@DzmitryHumianiuk i believe we can put it on hold since new version of client is coming soon.

**Comment by e-olga** [comment]
Added to RP backlog

**Comment by admizh** [comment]
Hi Team. I have same issue in RP v3.+

Steps to reproduce:
1. Put incorrect endpoint URL to reportportal.properties
`rp.endpoint = http://unknown-host:8080/`
2. Execute test and set brekpoins to **com.google.common.base.Throwables#throwIfUnchecked**
3. TestNg agent  should throw **com.epam.reportportal.restclient.endpoint.exception.RestEndpointIOException**: Unable to execute request - because this host not avalible

Current results:
Error not re-throwed,  no any messages in console (about that RP endpoint not avalible)
 
because `**RestEndpointIOException**` is not instanse of RuntimeException/Error
```
public static void throwIfUnchecked(Throwable throwable) {
    checkNotNull(throwable);
    if (throwable instanceof RuntimeException) {
      throw (RuntimeException) throwable;
    }
    if (throwable instanceof Error) {
      throw (Error) throwable;
    }
  }
```

**Comment by avarabyeu** [comment]
Fixed in 3.0
